Simulator/Recorders/RecordableBase.h

```diff
@@ -14,35 +14,45 @@
 #pragma once
 
 using namespace std;
+#include <typeinfo>
 #include <variant>
 #include <vector>
 
+/// A list of pre-defined basic data types for variablse in all the simulations
+/// These pre-defined types should match with the types of variant in Recorder
+using variantTypes = variant<uint64_t, bool, int, BGFLOAT>;
+
 class RecordableBase {
-   /// TODO: will change function names later
 public:
    virtual ~RecordableBase() = default;
```

Comment:
So, I did a little poking around. This class is fine as it is, in that if somehow someone wrote some code that instantiated a `RecordableBase` object, they will get a runtime error. However, if we want, we may be able to do one of the following to generate a compile-time error:

1. Declare the destructor as `virtual ~ RecordableBase() = 0;`, but then also provide the implementation outside of the class `RecordableBase::~ RecordableBase() {}`.
2. Declare a `protected` constructor.

The 1st option makes this an abstract base class; the latter prevents any code outside this class from creating a `RecordableBase` object. Supposedly.

Reply:
I chose the second solution. The first solution generates an error when serializing this class